### DISCUSSION TITLE
Add search functionality to `Listbox`

### DISCUSSION THREAD

**Discussion by dreamawakening** [tailwindlabs/headlessui#806] [Ideas]
It'd have similar behaviour to react-select: https://react-select.com/home

Specifically:

* The ability to type in the search box when selecting an option
* Async functionality for fetching options from a remote location

**Comment by iamadagostino** [comment]
It would be amazing to have such an important feature like this.

**Comment by hyunwoona** [comment]
Please add this functionality

**Comment by Chaphasilor** [comment]
To add to this, I would appreciate if this would also support something that `react-select` calls ["creatable" mode](https://react-select.com/home#creatable).  
Basically, suggest matching options, but also have an option to create a new item. This means adding another option ("create new") at the bottom of the list, that the user can choose instead of an existing option.

This would be the same behavior as when creating a new branch on GitHub, here's a quick demo:  

https://user-images.githubusercontent.com/18015147/140647127-3cd8c145-0530-4460-acc0-5baa20935cda.mp4

**Comment by ryanmichaelhirst** [comment]
+1 to adding this
